### PR TITLE
Add errors to windows.WSAStartup and WSACleanup

### PR DESCRIPTION
**Pull request by truemedian**

### CODE REVIEW

**Comment by andrewrk** [comment]
Thanks!

File: lib/std/os/windows.zig [hunk 1]
```diff
@@ -901,14 +901,23 @@ pub fn WSAStartup(majorVersion: u8, minorVersion: u8) !ws2_32.WSADATA {
     var wsadata: ws2_32.WSADATA = undefined;
     return switch (ws2_32.WSAStartup((@as(WORD, minorVersion) << 8) | majorVersion, &wsadata)) {
         0 => wsadata,
-        else => |err| unexpectedWSAError(@intToEnum(ws2_32.WinsockError, @intCast(u16, err))),
+        else => |err_int| switch (@intToEnum(ws2_32.WinsockError, @intCast(u16, err_int))) {
+            .WSASYSNOTREADY => return error.SystemNotAvailable,
+            .WSAVERNOTSUPPORTED => return error.VersionNotSupported,
+            .WSAEINPROGRESS => return error.BlockingOperationInProgress,
```
Comment (daurnimator) [inline lib/std/os/windows.zig:907]: I don't think this is actually possible?
Do you have some sample code to trigger the code path?

File: lib/std/os/windows.zig [hunk 2]
```diff
@@ -901,14 +901,23 @@ pub fn WSAStartup(majorVersion: u8, minorVersion: u8) !ws2_32.WSADATA {
     var wsadata: ws2_32.WSADATA = undefined;
     return switch (ws2_32.WSAStartup((@as(WORD, minorVersion) << 8) | majorVersion, &wsadata)) {
         0 => wsadata,
-        else => |err| unexpectedWSAError(@intToEnum(ws2_32.WinsockError, @intCast(u16, err))),
+        else => |err_int| switch (@intToEnum(ws2_32.WinsockError, @intCast(u16, err_int))) {
```
Comment (daurnimator) [inline lib/std/os/windows.zig:904]: this is a switch in a switch.....

Comment (andrewrk) [reply lib/std/os/windows.zig:904]: This pull request is already merged, can you submit a new one rather than reviewing it?

**Pull request merged**